### DISCUSSION TITLE
API Question - Is there an undocumented API for views?

### DISCUSSION THREAD

**Discussion by joelghill** [getredash/redash#6391] [Q&A]
We are currently using Redash to provide reports to users in our own custom UI. We are unable to embed the tables/views so we are using the API to fetch the results of queries and using that to populate our UI components. 

**Is there an API in the works for fetching the raw data for pre-made views**? It would be fantastic if we were able to compose a pie chart in Redash and then fetch the backing data via the API and render on the client side with UI components of our choosing.

**Comment by justinclift** [comment]
Reading over that, it's a bit unclear what the difference is between "results of queries" and "raw data for pre-made views".

Aren't they the same thing?

---

That being said, this part is interesting as well:

```
We are unable to embed the tables/views ...
```

Are you meaning that embedding charts isn't working for you?

Here's an example of an embedded Redash chart, in use for another project:

https://sqlitebrowser.org/stats/

Guessing you're meaning something else?
